### ISSUE TITLE
Allow copying drives, DVDs and other discs from the main view

### ISSUE THREAD

**Issue by Guntriaske457** [files-community/Files#4262]
**Is your feature request related to a problem?**
In Windows Explorer, the user can copy a DVD in the main view, where the drives are displayed, and paste it on the Desktop.

Windows Explorer automatically creates a main folder with the name of the DVD and then the internal data (files and folders) are copied.

**Describe the solution**
In Files have the possibility directly in the main view by mouse on a DVD select (Copy) and then paste on the desktop. Then Files automatically creates a main folder with the name of the DVD and then the internal data (files and folders) are copied into the main folder.

**Describe alternatives you've considered**
N/A

**Additional comment**
This should also work for CD and external drives (external hard drive). So if user for example clicks on an external hard drive (Copy) and on the Desktop (Paste). It creates a main folder with the name of the drive and then the internal data is copied.

**Comment by hecksmosis** [comment]
Should we implement this for drives, if so, what would be the expected functionality?

**Comment by hecksmosis** [comment]
I'll implement this when @Ferrario-Filippo implements context for widgets

**Comment by yaira2** [comment]
> Should we implement this for drives, if so, what would be the expected functionality? (A full copy of the drive contents/create a shortcut to it)

What does File Explorer do? We should be consistent with this.

**Comment by yaira2** [comment]
> Should we implement this for drives, if so, what would be the expected functionality? (A full copy of the drive contents/create a shortcut to it)

I would expect to copy the contents. We can add an option to paste as a shortcut but that's a different feature.

**Comment by hecksmosis** [comment]
For some reason, primary items don't appear in the sidebar context menu (I'm looking into it but help would be appreciated)

**Comment by yaira2** [comment]
@hecksmosis have you had any luck?

**Comment by ferrariofilippo** [comment]
> For some reason, primary items don't appear in the sidebar context menu (I'm looking into it but help would be appreciated)

Right-click context menu items are created manually in sidebar. Check out:
`GetLocationItemMenuItems(INavigationControlItem item, CommandBarFlyout menu)` at `Files.App.ViewModels.UserControls.SidebarViewModel`

**Comment by workbysaran** [comment]
@yaira2 

If I set the menu item as Primary, it doesn’t appear, so it is currently not set as Primary. I’ll do further research to identify the root cause.

**Comment by yair100** [comment]
Thank you for the update.

**Comment by workbysaran** [comment]
Fixed the primary button issue as well.

**Comment by yair100** [comment]
This feature will be included in the next update.
